### ISSUE TITLE
USB permission change came in twice causing unintended removal of entries in database

### ISSUE THREAD

**Issue by MoleMan1024** [MoleMan1024/audiowagon#46]
**Describe the issue**
seen by a user: Entries were removed from database around 16:12h in logfile:

```
2021-11-12 16:12:42.585 [DEBUG   ][22186][AudioItemRepo       ] Removing track from database: Track(trackId=11721, name='39, parentArtistId=403, parentAlbumId=1012, indexOnAlbum=11, albumArtURIString=, yearEpochTime=-1, genre=, uriString=usbAudio://5118-24576/Music%2FMusic%2FQueen%2FQueen%20-%20Greatest%20Hits%20IV%2F'39.mp3, lastModifiedEpochTime=1636539784403, durationMS=208909)
```

**Analysis**
Somehow two separate USB_PERMISSION_CHANGE events arrived, this might have caused this issue:

```
2021-11-12 16:11:52.611 [DEBUG   ][15376][USBDevConn          ] onReceive(de.moleman1024.audiowagon.AudioBrowserService@f9ed0c8; Intent { act=de.moleman1024.audiowagon.authorization.USB_PERMISSION_CHANGE flg=0x10 (has extras) })
...
2021-11-12 16:12:01.588 [DEBUG   ][15376][USBDevConn          ] onReceive(de.moleman1024.audiowagon.AudioBrowserService@f9ed0c8; Intent { act=de.moleman1024.audiowagon.authorization.USB_PERMISSION_CHANGE flg=0x10 (has extras) })
```

**AudioWagon version**
1.0.1 on Polestar 2

**Logfiles**
[audiowagon_2021-11-12_16-11-53.zip](https://github.com/MoleMan1024/audiowagon/files/7529751/audiowagon_2021-11-12_16-11-53.zip)

**Comment by MoleMan1024** [comment]
Yeah looks like two indexing threads were running in parallel, this should not happen.

**Comment by MoleMan1024** [comment]
Permission popup was also called twice: 
```
2021-11-12 16:11:47.288 [DEBUG   ][24679][USBDevConn          ] Waiting for user to grant permission
...
2021-11-12 16:11:49.697 [DEBUG   ][15376][USBDevConn          ] Waiting for user to grant permission
```
Probably related to timing of `android.intent.action.SCREEN_ON`

**Comment by MoleMan1024** [comment]
fixed in 1.1.0
